servers/common/common/neo4j/healthcheck.py

```diff
@@ -0,0 +1,181 @@
+import asyncio
+import sys
+import time
+
+from neo4j import AsyncDriver, Driver
+from neo4j.exceptions import DatabaseError
+
+
+async def attempt_neo4j_connection_async(
+    neo4j_driver: AsyncDriver,
+    database: str,
+    initial_delay: float = 3.0,
+    max_attempts: int = 3,
+    retry_delay_multiplier: float = 2.0,
+):
+    """Async function used to check the Neo4j database connection."""
+    attempts = 0
+    success = False
+    last_exception = DatabaseError()
+
+    # Initial delay
+    time.sleep(initial_delay)
+    print("\nWaiting for Neo4j to Start...\n", file=sys.stderr)
+
+    while not success and attempts < max_attempts:
```

Comment:
isn't that more like a test issue than a runtime issue? at runtime the neo4j db should be up and running

should be handled in test infrastructure instead that waits that a server is up (like testcontainers does internally)